scripts/customEventHooks.lua

```diff
@@ -1,39 +1,14 @@
-local customEventHooks = {}
-local template = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+local ScriptLoader = require('scriptLoader')
 
--- Variables for event registration and re-registration logging
-local previousValidatorEvents = {}
-local previousHandlerEvents = {}
-local previousValidatorFilePath = nil
-local previousHandlerFilePath = nil
-
-customEventHooks.validators = {}
-customEventHooks.handlers = {}
-customEventHooks.scriptID = {
-    handlers = {},
+---@class CustomEventHooks
+---@field validators table<string, table<string, function[]>> Map of event to source to callbacks
+---@field handlers table<string, table<string, function[]>> Map of event to source to callbacks
+local customEventHooks = {
     validators = {},
-    generatedScriptIDs = {}
+    handlers = {},
 }
 
-function customEventHooks.generateScriptID(filePath)
-    if not string.match(filePath, "%S") then
-        return nil
-    end
-
-    local seed = 0
-    for i = 1, #filePath do
-        local charCode = string.byte(filePath:normalizePath(), i)
```

Comment:
This shouldn't have filePath:normalizePath() anymore, it should have been sanitized before reaching this loop, if it wasn't it'd cause a range error.